src/boosting/prediction_early_stop.cpp

```diff
@@ -83,6 +83,9 @@ PredictionEarlyStopInstance CreatePredictionEarlyStopInstance(const std::string&
   } else {
     Log::Fatal("Unknown early stopping type: %s", type.c_str());
   }
+
+  // Fix for compiler warnings about reaching end of control
+  return CreateNone(config);
```

Comment:
maybe line 77 to 78 could be removed too?

Reply:
that would change the behavior of this function. This line I added should technically never be reached (since `Log::Fatal` will be called if an unrecognized `type` is passed).

If 77-78 was deleted, code that uses `type == "none"` would result in `Log::Fatal`

Reply:
thanks, got it.